webcompat/helpers.py

```diff
@@ -99,61 +99,72 @@ def get_user_info():
-    version = ua.get('major', u'Unknown')
-    # Add on the minor and patch version numbers if they exist
-    if version != u'Unknown' and ua.get('minor'):
-        version = version + "." + ua.get('minor')
-        if ua.get('patch'):
-            version = version + "." + ua.get('patch')
-    else:
-        version = ''
-    # Check for tablet devices
-    if ua_dict.get('device').get('model') == 'Tablet':
-        model = ' (Tablet)'
-    else:
-        model = ''
-    return '{0} {1}{2}'.format(name, version, model)
+    if user_agent_string and isinstance(user_agent_string, basestring):
+        ua_dict = user_agent_parser.Parse(user_agent_string)
+        ua = ua_dict.get('user_agent')
+        name = ua.get('family')
+        version = ua.get('major', u'Unknown')
+        # Add on the minor and patch version numbers if they exist
+        if version != u'Unknown' and ua.get('minor'):
+            version = version + "." + ua.get('minor')
+            if ua.get('patch'):
+                version = version + "." + ua.get('patch')
+        else:
+            version = ''
+        # Check for tablet devices
+        if ua_dict.get('device').get('model') == 'Tablet':
+            model = '(Tablet) '
+        else:
+            model = ''
+        rv = '{0} {1}{2}'.format(name, model, version)
+        # bizarre UA strings can be parsed like so:
+        # {'major': None, 'minor': None, 'family': 'Other', 'patch': None}
+        # but we want to return "Unknown", rather than "Other"
+        print(rv)
+        if rv.strip().lower() == "other":
+            return "Unknown"
+        return rv
+    return "Unknown"
+
+
+def get_browser_name(user_agent_string=None):
+    '''Return just the browser name.
+
+    unknown user agents will be reported as "unknown".
+    '''
+    if user_agent_string and isinstance(user_agent_string, basestring):
+        # get_browser will return something like 'Chrome Mobile 47.0'
+        # we just want 'chrome mobile', i.e., the lowercase name
+        # w/o the version
+        return get_browser(user_agent_string).rsplit(' ', 1)[0].lower()
+    return "unknown"
 
 
-def get_os(user_agent_string):
+def get_os(user_agent_string=None):
     '''Return operating system name.
 
     It pre-populates the bug reporting form.
     '''
-    ua_dict = user_agent_parser.Parse(user_agent_string)
-    os = ua_dict.get('os')
-    version = os.get('major', u'Unknown')
-    if version != u'Unknown' and os.get('major'):
-        version = version + "." + os.get('minor')
-        if os.get('patch'):
-            version = version + "." + os.get('patch')
-    else:
-        version = ''
-    return '{0} {1}'.format(os.get('family'), version)
+    if user_agent_string and isinstance(user_agent_string, basestring):
+        ua_dict = user_agent_parser.Parse(user_agent_string)
+        os = ua_dict.get('os')
+        version = os.get('major', u'Unknown')
+        if version != u'Unknown' and os.get('major'):
```

Comment:
this seems redundant here. If version is not unknown it means that `os.get('major')` in the previous line. so you don't need to test that again.